### PR TITLE
feat(extensions): added an option to disable related dependencies when disabling an extension with enabled dependencies

### LINKED ISSUE #102906: Extensions: Offer the user the ability to disable all dependencies when disabling an extension with dependents
Issue Type: <b>Bug</b>

Cannot activate the 'GitHub' extension because it depends on the 'Git' extension, which is disabled. Would you like to enable the extension and reload the window?

VS Code version: Code 1.47.2 (17299e413d5590b14ab0340ea477cdd86ff13daf, 2020-07-15T18:22:06.216Z)
OS version: Windows_NT x64 10.0.18363
Remote OS version: Linux x64 4.4.0-18362-Microsoft

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-7200U CPU @ 2.50GHz (4 x 2712)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: enabled<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|3.90GB (0.42GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|WSL: Ubuntu|
|OS|Linux x64 4.4.0-18362-Microsoft|
|CPUs|Intel(R) Core(TM) i5-7200U CPU @ 2.50GHz (4 x 2701)|
|Memory (System)|3.90GB (0.42GB free)|
|VM|0%|
</details><details><summary>Extensions (20)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-languagetool|ada|3.8.0
vscode-languagetool-en|ada|3.8.0
project-manager|ale|11.1.0
code-gnu-global|aus|0.2.2
gitignore|cod|0.6.0
git-extension-pack|don|0.1.3
githistory|don|0.6.8
gitlens|eam|10.2.2
code-runner|for|0.11.0
terminal|for|0.0.10
anaconda-extension-pack|ms-|1.0.1
python|ms-|2020.7.94776
cpptools|ms-|0.29.0
vscode-yaml|red|0.9.1
vscode-open-in-github|ziy|1.3.6
project-manager|ale|11.1.0
theme-monokai-pro-vscode|mon|1.1.17
remote-wsl|ms-|0.44.4
vscode-icons|vsc|10.2.0
material-theme|zhu|3.8.4

(9 theme extensions excluded)

</details>

### PR DESCRIPTION
**Pull request by rotem-bar**
This PR fixes #102906

example image

Adds a "Disable all related dependencies" button to the "Cannot disable extension X. Extensions .. depend on this"
To test changes: attempt to disable the Git extension while having an extension that depends on it enabled (Git Blame, etc.)

### CODE REVIEW

File: src/vs/workbench/contrib/extensions/browser/extensionsWorkbenchService.ts [hunk 1]
```diff
@@ -1050,7 +1050,19 @@ export class ExtensionsWorkbenchService extends Disposable implements IExtension
 			for (const extension of extensions) {
 				let dependents = this.getDependentsAfterDisablement(extension, allExtensions, this.local);
 				if (dependents.length) {
-					return Promise.reject(new Error(this.getDependentsErrorMessage(extension, allExtensions, dependents)));
+					return new Promise((resolve, reject) => {
+						this.notificationService.prompt(Severity.Error, this.getDependentsErrorMessage(extension, allExtensions, dependents), [
+							{
+								label: nls.localize('disable dep and dependents', 'Disable all related dependencies'),
```
Comment (gjsjohnmurray) [inline src/vs/workbench/contrib/extensions/browser/extensionsWorkbenchService.ts:1056]:
```suggestion
								label: nls.localize('disable extension and dependents', 'Disable all related dependencies'),
```

**Review by gjsjohnmurray — CHANGES_REQUESTED** [review]
I also suggest updating the three message texts in `getErrorMessageForDisablingAnExtensionWithDependents` to add the word "alone" like this:

`Cannot disable extension '{0}' alone.`

@sandy081 this is still on the August 2020 milestone, and I hope it can be merged soon. It will be particularly useful when disabling an extension that is part of an extension pack. At the moment we have to find the pack, disable that, then return to disable the member extension that we want to turn off.

**Comment by sandy081** [comment]
Sorry for not tracking this. Will go through it in November.

**Review by sandy081 — APPROVED** [review]
Thanks for the PR. Changes LGTM. 

There are some minor wording tweaks which I will do after merging.

**Pull request merged**